File: sdk/python/e2e_runner.py
import argparse
import os
import signal
import sys
import time

CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))
if CURRENT_DIR not in sys.path:
    sys.path.insert(0, CURRENT_DIR)

from slogx import slogx


def emit(prefix: str):
    slogx.info(f"{prefix} alpha", {"fixture": 1})
    slogx.warn(f"{prefix} beta", {"fixture": True})
    try:
        raise RuntimeError(f"{prefix} error")
    except Exception as err:
        slogx.error(f"{prefix} gamma", err)
    slogx.debug(f"{prefix} delta", {"nested": {"ok": True}})


def main():
    parser = argparse.ArgumentParser()
    parser.add_argument("--mode", default="ws", choices=["ws", "ci"])
    parser.add_argument("--port", type=int, default=8092)
    parser.add_argument("--file", default="")
    parser.add_argument("--service", default="py-e2e")
    parser.add_argument("--interval", type=float, default=0.2)
    args = parser.parse_args()

    slogx.init(
        is_dev=True,
        port=args.port,
        service_name=args.service,
        ci_mode=(args.mode == "ci"),
        log_file_path=args.file or None,
    )

    prefix = f"SDK PY {args.mode.upper()}"

    if args.mode == "ws":
        print(f"[slogx-e2e] READY ws://localhost:{args.port}", flush=True)
        while True:
            emit(prefix)
            time.sleep(args.interval)
    else:
        emit(prefix)
        time.sleep(0.65)
        emit(prefix)
        time.sleep(0.25)
        slogx.close()


def _handle_exit(_signum, _frame):
    slogx.close()
    sys.exit(0)


signal.signal(signal.SIGTERM, _handle_exit)
signal.signal(signal.SIGINT, _handle_exit)

if __name__ == "__main__":
    main()

File: sdk/rust/examples/e2e_runner.rs
use std::env;
use std::time::Duration;

fn get_arg_value(args: &[String], key: &str, default: &str) -> String {
    args.iter()
        .position(|arg| arg == key)
        .and_then(|idx| args.get(idx + 1))
        .cloned()
        .unwrap_or_else(|| default.to_string())
}

async fn emit(prefix: &str) {
    let alpha = format!("{} alpha", prefix);
    let beta = format!("{} beta", prefix);
    let gamma = format!("{} gamma", prefix);
    let delta = format!("{} delta", prefix);

    slogx::info!(alpha, { "fixture": 1 });
    slogx::warn!(beta, { "fixture": true });
    slogx::error!(gamma, { "error": format!("{} error", prefix) });
    slogx::debug!(delta, { "nested": { "ok": true } });
}

#[tokio::main]
async fn main() {
    let args: Vec<String> = env::args().collect();
    let mode = get_arg_value(&args, "--mode", "ws");
    let port: u16 = get_arg_value(&args, "--port", "8094")
        .parse()
        .unwrap_or(8094);
    let file_path = get_arg_value(&args, "--file", "");
    let service = get_arg_value(&args, "--service", "rust-e2e");
    let interval_ms: u64 = get_arg_value(&args, "--interval", "200")
        .parse()
        .unwrap_or(200);

    let ci_mode = mode == "ci";
    let log_file = if file_path.is_empty() { None } else { Some(file_path) };

    slogx::init_with_config(true, port, &service, Some(ci_mode), log_file, Some(1000)).await;

    let prefix = format!("SDK RUST {}", mode.to_uppercase());

    if mode == "ws" {
        println!("[slogx-e2e] READY ws://localhost:{}", port);
        let mut interval = tokio::time::interval(Duration::from_millis(interval_ms));
        loop {
            interval.tick().await;
            emit(&prefix).await;
        }
    }

    emit(&prefix).await;
    tokio::time::sleep(Duration::from_millis(650)).await;
    emit(&prefix).await;
    tokio::time::sleep(Duration::from_millis(150)).await;
}
